### PR TITLE
WW-175 Fix XSS vulnerability in appreciations() method

### PR DESCRIPTION
**Pull request by sebastianmarzjan**
- remove the vulnerable method
- use the rendering engine directly

### CODE REVIEW

File: extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php [hunk 1]
```diff
@@ -45,22 +45,18 @@ public function getAppreciations() {
...
 			if ( !empty( $appreciations ) ) {
-				$html = $this->app->renderView( 'ContributionAppreciation', 'appreciations', [
-					'appreciations' => $appreciations
-				] );
+				$numberOfAppreciations = count( $appreciations );
+				$numberOfHiddenAppreciations = $numberOfAppreciations > 2 ? $numberOfAppreciations - 2 : 0;
+				$html = ( new Wikia\Template\PHPEngine )
+					->setVal('appreciations', $appreciations)
+					->setVal('numberOfHiddenAppreciations', $numberOfHiddenAppreciations)
+					->render( __DIR__ . '/templates/ContributionAppreciation_appreciations.php' );
```
Comment (Grunny) [inline extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php]: Could this just be the template for `getAppreciations` instead?

Comment (sebastianmarzjan) [inline extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php]: Good point, that'll be way more straightforward.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 19 issues
* MAJOR 15 major
* MINOR 4 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ContributionAppreciationController.class.php#L27](https://github.com/Wikia/app/blob/58b26dccfc8f1d5684d6895375cec2d2a233d6e4/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L27): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
1. MAJOR [ContributionAppreciationController.class.php#L65](https://github.com/Wikia/app/blob/58b26dccfc8f1d5684d6895375cec2d2a233d6e4/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L65): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [ContributionAppreciationController.class.php#L82](https://github.com/Wikia/app/blob/58b26dccfc8f1d5684d6895375cec2d2a233d6e4/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L82): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [ContributionAppreciationController.class.php#L90](https://github.com/Wikia/app/blob/58b26dccfc8f1d5684d6895375cec2d2a233d6e4/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L90): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [ContributionAppreciationController.class.php#L100](https://github.com/Wikia/app/blob/58b26dccfc8f1d5684d6895375cec2d2a233d6e4/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L100): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [ContributionAppreciationController.class.php#L102](https://github.com/Wikia/app/blob/58b26dccfc8f1d5684d6895375cec2d2a233d6e4/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L102): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [ContributionAppreciationController.class.php#L104](https://github.com/Wikia/app/blob/58b26dccfc8f1d5684d6895375cec2d2a233d6e4/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L104): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [ContributionAppreciationController.class.php#L113](https://github.com/Wikia/app/blob/58b26dccfc8f1d5684d6895375cec2d2a233d6e4/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L113): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [ContributionAppreciationController.class.php#L129](https://github.com/Wikia/app/blob/58b26dccfc8f1d5684d6895375cec2d2a233d6e4/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L129): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1067)
1. MAJOR [ContributionAppreciationController.class.php#L134](https://github.com/Wikia/app/blob/58b26dccfc8f1d5684d6895375cec2d2a233d6e4/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L134): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

**Comment by Grunny** [comment]
LGTM.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 19 issues
- MAJOR 15 major
- MINOR 4 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [ContributionAppreciationController.class.php#L27](https://github.com/Wikia/app/blob/eec45a3877fea93bd46f423fdeb301f940109378/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L27): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
2. MAJOR [ContributionAppreciationController.class.php#L65](https://github.com/Wikia/app/blob/eec45a3877fea93bd46f423fdeb301f940109378/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L65): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
3. MAJOR [ContributionAppreciationController.class.php#L82](https://github.com/Wikia/app/blob/eec45a3877fea93bd46f423fdeb301f940109378/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L82): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
4. MAJOR [ContributionAppreciationController.class.php#L90](https://github.com/Wikia/app/blob/eec45a3877fea93bd46f423fdeb301f940109378/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L90): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
5. MAJOR [ContributionAppreciationController.class.php#L100](https://github.com/Wikia/app/blob/eec45a3877fea93bd46f423fdeb301f940109378/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L100): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
6. MAJOR [ContributionAppreciationController.class.php#L102](https://github.com/Wikia/app/blob/eec45a3877fea93bd46f423fdeb301f940109378/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L102): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
7. MAJOR [ContributionAppreciationController.class.php#L104](https://github.com/Wikia/app/blob/eec45a3877fea93bd46f423fdeb301f940109378/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L104): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
8. MAJOR [ContributionAppreciationController.class.php#L113](https://github.com/Wikia/app/blob/eec45a3877fea93bd46f423fdeb301f940109378/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L113): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
9. MAJOR [ContributionAppreciationController.class.php#L129](https://github.com/Wikia/app/blob/eec45a3877fea93bd46f423fdeb301f940109378/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L129): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1067)
10. MAJOR [ContributionAppreciationController.class.php#L134](https://github.com/Wikia/app/blob/eec45a3877fea93bd46f423fdeb301f940109378/extensions/wikia/ContributionAppreciation/ContributionAppreciationController.class.php#L134): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

**Comment by dianafa** [comment]
LGTM

**Pull request merged**